File: xpenbox-back/src/main/java/org/xpenbox/transaction/dto/TransactionResponseDTO.java
package org.xpenbox.transaction.dto;

import java.math.BigDecimal;

import org.xpenbox.account.dto.AccountResponseDTO;
import org.xpenbox.category.dto.CategoryResponseDTO;
import org.xpenbox.creditcard.dto.CreditCardResponseDTO;
import org.xpenbox.income.dto.IncomeResponseDTO;
import org.xpenbox.transaction.entity.Transaction.TransactionType;

import io.quarkus.runtime.annotations.RegisterForReflection;

/**
 * Data Transfer Object for responding with Transaction details.
 * @param resourceCode Resource code of the transaction.
 * @param description Description of the transaction.
 * @param transactionType Type of the transaction.
 * @param amount Amount involved in the transaction.
 * @param latitude Latitude where the transaction took place.
 * @param longitude Longitude where the transaction took place.
 * @param transactionDateTimestamp Timestamp of the transaction date and time.
 * @param category Associated category details.
 * @param income Associated income details.
 * @param account Associated account details.
 * @param creditCard Associated credit card details.
 * @param destinationAccount Associated destination account details (for transfers).
 */
@RegisterForReflection
public record TransactionResponseDTO (
    String resourceCode,
    String description,
    TransactionType transactionType,
    BigDecimal amount,
    BigDecimal latitude,
    BigDecimal longitude,
    Long transactionDateTimestamp,
    CategoryResponseDTO category,
    IncomeResponseDTO income,
    AccountResponseDTO account,
    CreditCardResponseDTO creditCard,
    AccountResponseDTO destinationAccount
) { }

File: xpenbox-front/src/app/feature/transaction/model/transactionResponseDTO.ts
import { AccountResponseDTO } from "../../account/model/accountResponseDTO";
import { CategoryResponseDTO } from "../../category/model/categoryResponseDTO";
import { CreditCardResponseDTO } from "../../creditcard/model/creditCardResponseDTO";
import { TransactionType } from "./transactionRequestDTO";

export interface TransactionResponseDTO {
    resourceCode: string;
    description: string;
    transactionType: TransactionType;
    amount: number;
    latitude?: number;
    longitude?: number;
    transactionDateTimestamp: number;
    category?: CategoryResponseDTO;
    income?: AccountResponseDTO;
    account?: AccountResponseDTO;
    creditCard?: CreditCardResponseDTO;
    destinationAccount?: AccountResponseDTO;
}
